src/main/java/redis/clients/jedis/ScanRoundRobin.java

```diff
@@ -0,0 +1,43 @@
+package redis.clients.jedis;
+
+import java.util.function.Function;
+import redis.clients.jedis.params.ScanParams;
+import redis.clients.jedis.Protocol.Keyword;
+import redis.clients.jedis.providers.ConnectionProvider;
+import redis.clients.jedis.resps.ScanResult;
+import redis.clients.jedis.util.JedisRoundRobinBase;
+
+public class ScanRoundRobin extends JedisRoundRobinBase<ScanResult<String>> {
+
+  private final int count;
+  private final Function<String, CommandArguments> args;
+
+  public ScanRoundRobin(ConnectionProvider connectionProvider, int batchCount, String match) {
```

Comment:
IMHO having the function being named the same as the class feels clunky. The class looks like it should be named RoundRobinScanner - or something, just to differentiate.

I recognize it's the norm (plenty of socket.socket in the universe), but  IMHO purpose based function/class combinations are more clear